core/src/main/scala/cats/syntax/all.scala

```diff
@@ -77,3 +78,5 @@ trait AllSyntaxBinCompat2
     with ValidatedSyntaxBincompat0
 
 trait AllSyntaxBinCompat3 extends UnorderedFoldableSyntax
+
+trait AllSyntaxBinCompat4 extends TraverseFilterSyntaxBinCompat0
```

Comment:
`AllSyntaxBinCompat3` was introduced since last release, you should be able to just use this.

Reply:
Ah, wonderful. I was actually thinking about that but hadn't checked before making this PR. I'll move the extension there, then.

I also plan on providing some test(s).